static/app/stores/teamStore.tsx

```diff
@@ -49,7 +63,11 @@ const teamStoreConfig: Reflux.StoreDefinition & TeamStoreInterface = {
     const item = this.getBySlug(itemId);
 
     if (!item) {
-      this.state.push(response);
+      this.state = {
+        ...this.state,
+        teams: [...this.state.teams, response],
```

Comment:
not sure if this is pretty typical for all our stores but these names feel a little generic. `response` seems to be an updated team object, but I can't really tell that from the name `response`. It's out of scope for this PR but might be worth updating later